blog/tanstack-router-intro/index.md

```diff
@@ -0,0 +1,465 @@
+---
+title: Introducing TanStack Router
+date: "2024-09-01T10:00:00.000Z"
+description: An introduction to TanStack Router
+---
+
+TanStack Router is an incredibly exciting project. It's essentially a fully-featured _client-side_ JavaScript application framework. It provides a mature routing and navigation system, with nested layouts, and efficient data loading capabilities at every point in the route tree. Best of all, it does all of this in a _type-safe_ manner.
+
+What's especially exciting is that as of this writing, there's a TanStack Start in the works, which will add server-side capabilities to Router, enabling you to build full-stack web applications. Start promises to do this with a server layer applied directly on top of the same TanStack Router we'll be talking about here. That makes this a perfect time to get to know Router if you haven't already.
+
+Router is much more than just a ... router; as we mentioned, it's a full-fledged client-side application framework. So to prevent this post from getting too long, we won't even try to cover it all. Here we'll limit ourselves to routing and navigation, which is a larger topic than you might think, especially considering the type-safe nature of Router.
+
+## Getting started
+
+The TanStack Router docs [are here](https://tanstack.com/router/latest/docs/framework/react/overview), and the quick start guide [is here](https://tanstack.com/router/latest/docs/framework/react/quick-start). To be frank, the quick start is a bit more manual, and less quick than it could be. For now. TanStack Start looks to add the same kind of nice scaffolding projects that like SvelteKit and Next have. But for now you can just clone [the repo](https://github.com/arackaf/tanstack-router-routing-demo) I used for this project and follow along.
```

Comment:
we now have scaffolding via e.g. `pnpm create @tanstack/router`
check the quickstart guide

https://tanstack.com/router/latest/docs/framework/react/quick-start#scaffolding-your-first-tanstack-router-project